### PR TITLE
Speed - for discussion

### PR DESCRIPTION
**Pull request by mbriskar**
1) Introduces  javaclass-ignore
         1.a registered ignore regexes are aplied during .class pre-scan
         1.b it uses trie structure (here I am not sure if this is not overengineered)
2.) TypeInterestFactory uses internally trie structure to answer queries. 

This needs to be properly tested

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/436/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1066/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/437/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/437/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/437/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1067/
Test PASSed.

File: java-ast/addon/src/main/java/org/jboss/windup/ast/java/ReferenceResolvingVisitor.java [hunk 1]
```diff
@@ -111,7 +113,7 @@ private void resolveCurrentTypeNames(CompilationUnit compilationUnit)
             {
                 typeLine = typeLine.substring(typeDeclaration.getJavadoc().toString().length());
             }
-            classReferences.add(new ClassReference(fqcn, ResolutionStatus.RESOLVED, TypeReferenceLocation.TYPE,
```
Comment (jsight) [inline java-ast/addon/src/main/java/org/jboss/windup/ast/java/ReferenceResolvingVisitor.java]: This seems to be reverting some changes. Is this the result of a bad rebase or something?

Comment (mbriskar) [inline java-ast/addon/src/main/java/org/jboss/windup/ast/java/ReferenceResolvingVisitor.java]: isn't this your change?

Comment (jsight) [inline java-ast/addon/src/main/java/org/jboss/windup/ast/java/ReferenceResolvingVisitor.java]: lol, apparently I am a little off today. Yes, that is my change and wasn't removing what I thought it was. I just misread it.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1068/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/438/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/438/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/438/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1069/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1072/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/441/
Test PASSed.

**Pull request merged**